File: go/maximum_matrix_sum.go
package main

import "math"

func maxMatrixSum(matrix [][]int) int64 {
	neg, sum, low := 0, 0, math.MaxInt
	for _, r := range matrix {
		for _, x := range r {
			if x < 0 {
				neg++
				sum -= x
				low = min(low, -x)
			} else {
				sum += x
				low = min(low, x)
			}
		}
	}
	return int64(sum - neg%2*low*2)
}

File: python/maximum_matrix_sum.py
class Solution:
    def maxMatrixSum(self, matrix: list[list[int]]) -> int:
        sum, neg, low = 0, 0, 1e9
        for row in matrix:
            for x in row:
                if x < 0:
                    neg += 1
                    x = -x
                sum += x
                low = min(low, x)
        if neg % 2 == 0:
            return sum
        return sum - 2 * low
